### PR TITLE
feat(cms): added product pages

### PR DESCRIPTION
**Pull request by nicolaisirghi**
### Changelog
- Added product page in CMS
- Added preview link in CMS for collections

### CODE REVIEW

File: packages/demo-nextjs/src/components/Metronic/modules/apps/products/components/table/columns/ProductActionsCell.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  id: string;
+  previewUrl: string;
+}
+
+const extractProductId = (url: string) => {
+  const parts = url.split("/");
+  return parts[parts.length - 1];
+};
+
+const SHOPIFY_BASE_URL = "https://admin.shopify.com/store";
```
Comment (maxval1) [inline packages/demo-nextjs/src/components/Metronic/modules/apps/products/components/table/columns/ProductActionsCell.tsx:18]: move inside `utils/mock.ts`

File: packages/demo-nextjs/src/components/Metronic/modules/apps/products/core/ProductsProvider.tsx [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  const { defaultProductPageUrl, isLoading: isFetchingDefaultProductPage } = useDefaultProductPageUrl();
+  const { totalItems, isLoading: isFetchingProductsCount } = useFetchProductsCount();
+
+  useEffect(() => {
+    refetchProducts();
+  }, [currentPage, refetchProducts]);
+
+  const handleEditProduct = useMutation(async (id: string) => {
+    try {
+      const collectionType = "product";
```
Comment (maxval1) [inline packages/demo-nextjs/src/components/Metronic/modules/apps/products/core/ProductsProvider.tsx:29]: use enum

File: packages/demo-nextjs/src/components/Metronic/modules/apps/products/core/_requests.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import { ItemsPerPage } from "@/components/Metronic/helpers";
+import { getConfig } from "@/config";
+import axios from "axios";
+
+const API_URL = `${getConfig().host}/api`;
```
Comment (maxval1) [inline packages/demo-nextjs/src/components/Metronic/modules/apps/products/core/_requests.ts:5]: remove `getConfig().host` use `/api`

**Review by maxval1 — CHANGES_REQUESTED** [review]

File: packages/demo-nextjs/src/app/[[...slug]]/page.tsx [hunk 4]
```diff
@@ -10,7 +10,7 @@ import { footerQuery, headerQuery, projectId } from "@/utils/mock";
 import { isT } from "fp-utilities";
 import { notFound } from "next/navigation";
 
-const excludeCatchAllCollection = [CollectionTypes.system, CollectionTypes.popup, CollectionTypes.story];
+const excludeCatchAllCollection = [ CollectionTypes.popup, CollectionTypes.story];
```
[the remark this answers is not in the document]
Comment (nicolaisirghi) [reply packages/demo-nextjs/src/app/[[...slug]]/page.tsx:13]: If I correctly understand the usage of the `excludeCatchAllCollection`, is for exclude the preview for items with the specific collection.

 Now, it also needs to include the system page for Product in the preview.
image

File: packages/demo-nextjs/src/app/admin/(editor)/product/[slug]/page.tsx [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+import React from "react";
+
+interface Props {
+  params: { slug: string };
+}
+
+export default async function ProductPage(props: Props) {
+  const { params } = props;
+  const slug = params.slug;
+  const pagePreview = `${getConfig().host}/product/${slug}`;
```
Comment (maxval1) [inline packages/demo-nextjs/src/app/admin/(editor)/product/[slug]/page.tsx:15]: image

**Review by maxval1 — CHANGES_REQUESTED** [review]

**Review by maxval1 — CHANGES_REQUESTED** [review]
Conflcits

**Pull request merged**